API/Services/ImageServices/ImageMagick/ImageMagickImage.cs

```diff
@@ -0,0 +1,227 @@
+    {
+        _image = new MagickImage(stream);
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="ImageMagickImage"/> from an existing <see cref="MagickImage"/>.
+    /// </summary>
+    /// <param name="image">The existing <see cref="MagickImage"/>.</param>
+    public ImageMagickImage(MagickImage image)
+    {
+        _image = (MagickImage)image.Clone();
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="ImageMagickImage"/> with the specified width, height, and color.
+    /// </summary>
+    /// <param name="width">The width of the image.</param>
+    /// <param name="height">The height of the image.</param>
+    /// <param name="red">The red component of the color.</param>
+    /// <param name="green">The green component of the color.</param>
+    /// <param name="blue">The blue component of the color.</param>
+    public ImageMagickImage(int width, int height, byte red = 0, byte green = 0, byte blue = 0)
+    {
+        _image = new MagickImage(MagickColor.FromRgb(red, green, blue), width, height);
+    }
+
+    /// <inheritdoc/>
+    public IImage Clone()
+    {
+        return new ImageMagickImage(_image);
+    }
+
+    /// <inheritdoc/>
+    public void Resize(int width, int height)
+    {
+        _image.Resize(width, height);
+    }
+
+    /// <inheritdoc/>
+    public void Crop(int x, int y, int width, int height)
+    {
+        _image.Crop(new MagickGeometry(x, y, width, height));
+    }
+
+    /// <inheritdoc/>
+    public void Thumbnail(int width, int height)
+    {
+        _image.Thumbnail(new MagickGeometry(width, height) { IgnoreAspectRatio = true });
+    }
+
+    /// <inheritdoc/>
+    public void Composite(IImage overlay, int x, int y)
+    {
+        ImageMagickImage tile = overlay as ImageMagickImage;
+        if (tile == null) return;
+        _image.Composite(tile._image, x, y, CompositeOperator.Over);
+    }
+
+    /// <inheritdoc/>
+    public void Save(string filename, EncodeFormat format, int quality)
+    {
+        _image.Quality = quality;
+        _image.Write(filename, MagickFormatFromEncodeFormat(format));
+    }
+
+    /// <inheritdoc/>
+    public void Save(Stream stream, EncodeFormat format, int quality)
+    {
+        _image.Quality = quality;
+        _image.Write(stream, MagickFormatFromEncodeFormat(format));
+    }
+
+    /// <inheritdoc/>
+    public Task SaveAsync(string filename, EncodeFormat format, int quality, CancellationToken token = default)
+    {
+        _image.Quality = quality;
+        return _image.WriteAsync(filename, MagickFormatFromEncodeFormat(format), token);
+    }
+
+    /// <inheritdoc/>
+    public Task SaveAsync(Stream stream, EncodeFormat format, int quality, CancellationToken token = default)
+    {
+        _image.Quality = quality;
+        return _image.WriteAsync(stream, MagickFormatFromEncodeFormat(format), token);
+    }
+
+    /// <inheritdoc/>
+    public float[] GetRGBAImageData()
+    {
+        float[] data = null;
```

Comment:
This either needs to be an empty array or needs nullable annotation.

Reply:
Done